File: sdk/python/hmac_send.py
import os, hmac, hashlib, uuid, json, requests, datetime
url=os.environ["N8N_WEBHOOK_URL"]
secret=os.environ["N8N_WEBHOOK_SECRET"].encode()
payload=json.dumps({"event":"demo","payload":{"x":1}})
ts=datetime.datetime.utcnow().isoformat()+"Z"
toSign=(ts+"."+payload).encode()
sig=b"sha256="+hmac.new(secret,toSign,hashlib.sha256).hexdigest().encode()
res=requests.post(url,headers={"Content-Type":"application/json","X-Signature":sig,"X-Timestamp":ts,"Idempotency-Key":str(uuid.uuid4())},data=payload)
print(res.status_code,res.text)

File: sdk/node/hmac_send.js
import crypto from "crypto";
import fetch from "node-fetch";
const url = process.env.N8N_WEBHOOK_URL;
const secret = process.env.N8N_WEBHOOK_SECRET;
(async ()=>{
  const payload=JSON.stringify({ event:"demo", payload:{ x:1 }});
  const ts=new Date().toISOString();
  const sig="sha256="+crypto.createHmac("sha256",secret).update(ts+"."+payload).digest("hex");
  const res=await fetch(url, {
    method:"POST",
    headers:{
      "Content-Type":"application/json",
      "X-Signature":sig,
      "X-Timestamp":ts,
      "Idempotency-Key":crypto.randomUUID()
    },
    body:payload
  });
  console.log(res.status,await res.text());
})();
